x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/support/ShapeValuesSource.java

```diff
@@ -16,8 +16,8 @@
 
 import java.io.IOException;
 
-public abstract class ShapeValuesSource<T extends ShapeValues<?>> extends ValuesSource {
-    public abstract T shapeValues(LeafReaderContext context);
+public abstract class ShapeValuesSource extends ValuesSource {
+    public abstract ShapeValues shapeValues(LeafReaderContext context);
```

Comment:
I disagree here, please do not remove the generic. GeoShapeValuesSource should return GeoShapeValues.